src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/UpdateAzureContainerRegistryCredential.cs

```diff
@@ -0,0 +1,58 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Management.ContainerRegistry.Models;
+
+namespace Microsoft.Azure.Commands.ContainerRegistry
+{
+    [Cmdlet(VerbsData.Update, ContainerRegistryCredentialNoun, SupportsShouldProcess = true),
+        OutputType(typeof(PSContainerRegistryCredential))]
+    public class UpdateAzureContainerRegistryCredential : ContainerRegistryCmdletBase
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias, RegistryNameAlias, ResourceNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of password to regenerate. " + AllowedPasswordNames)]
+        [ValidateNotNullOrEmpty]
+        public PasswordName PasswordName { get; set; }
+
```

Comment:
@djyou we try to push towards piping by object instead of by property name as often as possible. A good example would be if you had a complex object with a large number of properties that you were interesting in knowing (possibly for a PUT update cmdlet). It wouldn't be reasonable to have a parameter for _every single property_ that you were interested in and giving them the `ValueFromPipelineByPropertyName` attribute; this would be a bad experience for users who weren't piping the object, and it would take a little longer to execute as PowerShell would need to resolve the parameter bindings. Instead, if you accepted the complex object from the pipeline, you would only need one parameter, and it would take less time to resolve. Let me know if that answers your question.